### ISSUE TITLE
Redis "generation" control

### ISSUE THREAD

**Issue by dbutenhof** [distributed-system-analysis/pbench#2668]
We've seen problems where Redis isn't cleaned up during tool meister exit which cause Redis startup during a subsequent command to fail. This isn't diagnosed, but doesn't actually work due to the key setup.

First off we need to be better about shutting down on termination (especially in error cases, including SIGINT), which would mitigate the problem (e.g., #2625) ... but we also need to be able to diagnose this cleanly and continue.

One idea is to initialize our Redis connection by populating a known "tool meister claim" key with a value including something that's trackable to the tool meister master instance along with a UUID. When we connect to a Redis instance we expect that key not to exist, or to already have the controller's unique key (depending on context). We can then diagnose "we're seeing a Redis instance set up by some other tool meister instance" and take appropriate action.

**Comment by webbnh** [comment]
Do we assume that a given Redis server will never be used concurrently by more than one Tool Meister instance?

I suppose, if the Redis server was _created_ by a TM, then it probably shouldn't be shared with another, since the creator will presumably shut it down when it finishes, so the mechanism that you outline above makes good sense; however, for the other case, we should presumably skip all of that.

**Comment by dbutenhof** [comment]
Yes, if we're pointed at an external Redis server it'd probably be good to be able to share it between distinct Pbench controllers with their own TMS.
